os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/os/tizenrt/osif_tizenrt.c

```diff
@@ -544,11 +544,15 @@ bool osif_msg_send(void *p_handle, void *p_msg, uint32_t wait_ms)
 		return _FAIL;
 	}
 
-	if (osif_task_context_check() == true && wait_ms != 0xFFFFFFFF) {
+	if (osif_task_context_check() == true && wait_ms != 0xFFFFFFFF && wait_ms != 0) {

```

Comment:
How about checking `wait_ms` first?
This condition should check three cases, but if we check `wait_ms` first, then we can reduce the time if `wait_ms` is 0xFFFFFFFF or 0.